### PR TITLE
Fixing tokens displayed when sending an ALL_TOKEN transaction

### LINKED ISSUE #1838: "Send all [asset]" wants to send unrelated assets too
Hi everyone,

Trying to send some assets between wallets, there might be a bug when using the the 'Send all" checkbox.

When checking the "Send all [asset]" for a given transaction 
IMG_DDF1AAF20D09-1
 
The wallet shows a warning, however displays that it wants to send all the assets in the wallet
sending all hosk

The feature works properly with the browser extension.

### PR DESCRIPTION
**Pull request by zuzunker**
Added check for send-all AND default-asset-selected

### CODE REVIEW

**Review by Nebyt — APPROVED** [review]
Checked. Fixed. :rocket:

**Comment by wolverineks** [comment]
what is this change for?

File: legacy/components/Send/SendScreen.js [hunk 1]
```diff
@@ -259,25 +259,23 @@ class SendScreenLegacy extends Component<LegacyProps, State> {
 
       const fee = (await transactionData.fee()).getDefault()
 
-      const defaultAssetAmount = selectedTokenMeta.isDefault
+      let isDefaultAssetSelected = selectedTokenMeta.isDefault;
```
Comment (wolverineks) [inline legacy/components/Send/SendScreen.js]: why is this `let`?

**Comment by wolverineks** [comment]
> what is this change for?

it looks like this is making a list of tokens to display on the confirm screen
we should have a single source of truth
is there any way we can derive this from the transaction data itself?
what else can we derive from the transactionData?

```ts
const tokens: Array<{ tokenId: string, amount: string, address: string }> = getTokens(transactionData)
const fee: string = await getFee(transactionData)
```

**Comment by Nebyt** [comment]
> what is this change for?

It is for this https://github.com/Emurgo/yoroi-mobile/issues/1838 issue

**Review by wolverineks — CHANGES_REQUESTED** [review]
change `let` -> `const`
run the changes through prettier

File: legacy/components/Send/SendScreen.js (hunk 1 above)
Comment (zuzunker) [reply legacy/components/Send/SendScreen.js]: ah, automatically generated, haven't noticed. Good catch!

**Comment by zuzunker** [comment]
> change `let` -> `const` run the changes through prettier

@wolverineks, Updated. Runned it through `yarn prettify`

**Review by wolverineks — APPROVED** [review]

**Comment by wolverineks** [comment]
> > change `let` -> `const` run the changes through prettier
> 
> @wolverineks, Updated. Runned it through `yarn prettify`

just realized that `yarn prettify` doesnt format the old code, so you have to manually run prettier on that file

**Comment by zuzunker** [comment]
Weird, I have runned it through the prettify but now it fails with some prettier errors 🤔

**Comment by zuzunker** [comment]
> > > change `let` -> `const` run the changes through prettier
> > 
> > 
> > @wolverineks, Updated. Runned it through `yarn prettify`
> 
> just realized that `yarn prettify` doesnt format the old code, so you have to manually run prettier on that file

ahhhhhh, that explains it

**Review by zuzunker — COMMENTED** [review]
/check

**Pull request merged**